src/lib/controllers/ui-controller.js

```diff
@@ -1051,4 +1103,117 @@ export default class UIController {
+      TextSelector conveys text selection information. It is more generic of the two.
+      HTMLSelector conveys page-specific information, such as location of a selection on a page.
+      It's probably better to keep them separated in order to follow a more abstract model.
+       */
+      let htmlSelector = new HTMLSelector(event, this.options.items.preferredLanguage.currentValue)
+      let textSelector = htmlSelector.createTextSelector()
+
+      if (!textSelector.isEmpty()) {
+        // TODO: disable experience monitor as it might cause memory leaks
+        /* ExpObjMon.track(
+          LexicalQuery.create(textSelector, {
+            htmlSelector: htmlSelector,
+            uiController: this.ui,
+            maAdapter: this.maAdapter,
+            lexicons: Lexicons,
+            resourceOptions: this.resourceOptions,
+            siteOptions: [],
+            lemmaTranslations: this.enableLemmaTranslations(textSelector) ? { adapter: LemmaTranslations, locale: this.options.items.locale.currentValue } : null,
+            langOpts: { [Constants.LANG_PERSIAN]: { lookupMorphLast: true } } // TODO this should be externalized
+          }),
+          {
+            experience: 'Get word data',
+            actions: [
+              { name: 'getData', action: ExpObjMon.actions.START, event: ExpObjMon.events.GET },
+              { name: 'finalize', action: ExpObjMon.actions.STOP, event: ExpObjMon.events.GET }
+            ]
+          })
+          .getData() */
+
+        LexicalQuery.create(textSelector, {
+          htmlSelector: htmlSelector,
+          uiController: this,
+          maAdapter: this.maAdapter,
+          lexicons: Lexicons,
+          resourceOptions: this.resourceOptions,
+          siteOptions: [],
+          lemmaTranslations: this.enableLemmaTranslations(textSelector) ? { adapter: LemmaTranslations, locale: this.options.items.locale.currentValue } : null,
+          langOpts: { [Constants.LANG_PERSIAN]: { lookupMorphLast: true } } // TODO this should be externalized
+        }).getData()
+      }
+    }
+  }
+
+  /**
+   * Check to see if Lemma Translations should be enabled for a query
+   *  NB this is Prototype functionality
+   */
+  enableLemmaTranslations (textSelector) {
+    return textSelector.languageID === Constants.LANG_LATIN &&
+      this.options.items.enableLemmaTranslations.currentValue &&
+      !this.options.items.locale.currentValue.match(/^en-/)
+  }
+
+  handleEscapeKey (event) {
+    // TODO: Move to keypress as keyCode is deprecated
+    // TODO: Why does it not work on initial panel opening?
+    if (event.keyCode === 27 && this.state.isActive()) {
+      if (this.state.isPanelOpen()) {
+        this.panel.close()
+      } else if (this.popup.visible) {
+        this.popup.close()
+      }
+    }
+    return true
+  }
+
+  handleReload () {
+    console.log('Alpheios reload event caught.')
+    if (this.state.isActive()) {
+      this.deactivate()
+    }
+    window.location.reload()
```

Comment:
see note above, this event is specific to the reload of the webextension upon installation. not sure it belongs here.

Reply:
Think the same, will move it to the content script.